drivers/ethernet/Kconfig.stm32_hal

```
@@ -45,6 +45,13 @@ config ETH_STM32_HAL_RX_THREAD_STACK_SIZE
 	help
 	  RX thread stack size
 
+config ETH_STM32_HAL_RX_THREAD_PREEMPTIVE
+	int "STM32 Ethernet RX Thread pre-emptive [EXPERIMENTAL]"
+	depends on PREEMPT_ENABLED
+	select EXPERIMENTAL
+	help
+	  With pre-emptive threads, the thread can be pre-empted.
```

Should be a boolean, and make it default to the original value, no need to set this experimental IMO.

```suggestion
config ETH_STM32_HAL_RX_THREAD_PREEMPTIVE
	bool "STM32 Ethernet RX Thread pre-emptive"
	default y if NET_TC_THREAD_PREEMPTIVE
	depends on PREEMPT_ENABLED
	help
	  With pre-emptive threads, the thread can be pre-empted.
```

---

Good catch on the bool.

Left it experimental because original (TC_PREEMPTIVE) was experimental. I am not aware of whether this thread being preemtable is experimental or not.

---

Fixed type from `int` to `bool`.

---

Added the default clause.